cmd/process-agent/config_test.go

```diff
@@ -0,0 +1,111 @@
+)
+
+func TestProcessDiscovery(t *testing.T) {
+	scfg, ocfg := &sysconfig.Config{}, &oconfig.OrchestratorConfig{}
+
+	cfg := config.Mock()
+
+	// Make sure the process_discovery check can be enabled
+	cfg.Set("process_config.process_discovery.enabled", true)
+	enabledChecks := getChecks(scfg, ocfg, false)
+	assert.Contains(t, enabledChecks, checks.ProcessDiscovery)
+
+	// Make sure the process_discovery check can be disabled
+	cfg.Set("process_config.process_discovery.enabled", false)
+	enabledChecks = getChecks(scfg, ocfg, true)
+	assert.NotContains(t, enabledChecks, checks.ProcessDiscovery)
+
+	// Make sure the process and process_discovery checks are mutually exclusive
+	cfg.Set("process_config.process_discovery.enabled", true)
+	cfg.Set("process_config.process_collection.enabled", true)
+	enabledChecks = getChecks(scfg, ocfg, true)
+	assert.NotContains(t, enabledChecks, checks.ProcessDiscovery)
+}
+
+func TestContainerCheck(t *testing.T) {
+	scfg, ocfg := &sysconfig.Config{}, &oconfig.OrchestratorConfig{}
+
+	cfg := config.Mock()
+
+	// Make sure the container check can be enabled if the process check is disabled
+	cfg.Set("process_config.process_collection.enabled", false)
+	cfg.Set("process_config.container_collection.enabled", true)
+	enabledChecks := getChecks(scfg, ocfg, true)
+	assert.Contains(t, enabledChecks, checks.Container)
+	assert.Contains(t, enabledChecks, checks.RTContainer)
+	assert.NotContains(t, enabledChecks, checks.Process)
+
+	// Make sure that disabling RT disables the rt container check
+	cfg.Set("process_config.disable_realtime_checks", true)
+	enabledChecks = getChecks(scfg, ocfg, true)
+	assert.Contains(t, enabledChecks, checks.Container)
+	assert.NotContains(t, enabledChecks, checks.RTContainer)
+
+	// Make sure the container check cannot be enabled if we cannot access containers
+	enabledChecks = getChecks(scfg, ocfg, false)
+	assert.NotContains(t, enabledChecks, checks.Container)
+
+	// Make sure the container and process check are mutually exclusive
+	cfg.Set("process_config.process_collection.enabled", true)
+	cfg.Set("process_config.container_collection.enabled", true)
+	enabledChecks = getChecks(scfg, ocfg, true)
+	assert.Contains(t, enabledChecks, checks.Process)
+	assert.NotContains(t, enabledChecks, checks.Container)
+	assert.NotContains(t, enabledChecks, checks.RTContainer)
+}
+
+func TestProcessCheck(t *testing.T) {
+	cfg := config.Mock()
+	cfg.Set("system_probe_config.enabled", true)
+	cfg.Set("system_probe_config.process_config.enabled", true)
+
+	scfg, err := sysconfig.New("")
+	assert.NoError(t, err)
+
+	ocfg := &oconfig.OrchestratorConfig{}
+
+	// Make sure the process check can be enabled
+	cfg.Set("process_config.process_collection.enabled", true)
+	enabledChecks := getChecks(scfg, ocfg, true)
+	assert.Contains(t, enabledChecks, checks.Process)
+
+	// Make sure that the process check recognizes the system probe is available
+	assert.True(t, checks.Process.SysprobeProcessModuleEnabled)
```

Comment:
should we also test that it's set to false when the process module in sysprobe is disabled?

Reply:
Fixed! I overhauled these tests rather significantly, give it a look!

Reply:
They're great! Thanks for the update!